tests/python/gpu/test_kvstore_gpu.py

```diff
@@ -26,20 +26,20 @@
 str_keys = ['b', 'c', 'd']
 
 
-def init_kv_with_str(stype='default'):
+def init_kv_with_str(stype='default', kv_type='local'):
     """init kv """
-    kv = mx.kv.create()
+    kv = mx.kv.create(kv_type)
     # single
     kv.init('a', mx.nd.zeros(shape, stype=stype))
     # list
     kv.init(str_keys, [mx.nd.zeros(shape=shape, stype=stype)] * len(keys))
     return kv
 
 
-def test_row_sparse_pull():
-    kv = init_kv_with_str('row_sparse')
+def test_row_sparse_pull(kv_type='local'):
```

Comment:
I think nosetest will not run the code in `__main__` but rather look for functions starting with `test_`. Shall we test both `local` and `device` in `test_row_sparse_pull`?

Reply:
On second thought, we should have a separate test for kv=device with rsp values on different contexts. Using gpu ctxes is fine, CI should have more than 1 GPUs.